### ISSUE TITLE
Unable to detect rotated faces / landmarks inaccurate

### ISSUE THREAD

**Issue by xsacha** [supernotman/RetinaFace_Pytorch#4]
![t1](https://user-images.githubusercontent.com/61218/62817310-a24d7b80-bb77-11e9-98d9-3dffa68950ed.jpg)
![t4](https://user-images.githubusercontent.com/61218/62817311-a4afd580-bb77-11e9-97ff-8c96416cd98a.jpg)

These two test images look better in other implementations of RetinaFace for PyTorch, for eg.
https://github.com/bogireddytejareddy/retinaface-pytorch/blob/master/test_results/t1.jpg
https://github.com/bogireddytejareddy/retinaface-pytorch/blob/master/test_results/t4.jpg

**Comment by supernotman** [comment]
> ![t1](https://user-images.githubusercontent.com/61218/62817310-a24d7b80-bb77-11e9-98d9-3dffa68950ed.jpg)
> ![t4](https://user-images.githubusercontent.com/61218/62817311-a4afd580-bb77-11e9-97ff-8c96416cd98a.jpg)
> 
> These two test images look better in other implementations of RetinaFace for PyTorch, for eg.
> https://github.com/bogireddytejareddy/retinaface-pytorch/blob/master/test_results/t1.jpg
> https://github.com/bogireddytejareddy/retinaface-pytorch/blob/master/test_results/t4.jpg


> ![t1](https://user-images.githubusercontent.com/61218/62817310-a24d7b80-bb77-11e9-98d9-3dffa68950ed.jpg)
> ![t4](https://user-images.githubusercontent.com/61218/62817311-a4afd580-bb77-11e9-97ff-8c96416cd98a.jpg)
> 
> These two test images look better in other implementations of RetinaFace for PyTorch, for eg.
> https://github.com/bogireddytejareddy/retinaface-pytorch/blob/master/test_results/t1.jpg
> https://github.com/bogireddytejareddy/retinaface-pytorch/blob/master/test_results/t4.jpg

Thanks for your test.And i noticed the problem as well , i am still trying to figure it out and try to make the performance and recall for rotated faces better.
And i think there is some  probable cause for this:
1. I use different loss weight for cls,box and landmarks . In the paper,the author said  the loss-balancing parameters are set to 1, 0.25, 0.1,but i see the origin code it seems to be 1,1,0.5(I am not sure).So this might influence the performance.
2. I find that when i use random flip Augmentation, losses for landmarks seems weird like this:
![TIM截图20190812103320](https://user-images.githubusercontent.com/31950296/62843489-b8c91380-bcec-11e9-8e31-6e9cd2a4b43a.png)
Some noises in landmark loss.But i checked the groundtruth and found no wrong label.
3. There are some components to do like deformable convolution and data augumentation ways i listed in Todo list in readme. These may also make some affect. 
Still there are some bugs in the code and i am trying to fixed it. And it is welcomed to commit a issue if you have any find.
Thanks.

**Comment by xsacha** [comment]
Landmarks are more accurate after your latest bug fix! Getting high precision (over 90%) quite early now at about epoch 10.
Rotated faces are still not detected but I suppose that is due to the data loader.

**Comment by xsacha** [comment]
Just a note that epoch 39 was actually the best epoch I had out of the lot, so it only got worse from there.

**Comment by supernotman** [comment]
> Just a note that epoch 39 was actually the best epoch I had out of the lot, so it only got worse from there.

yes,I also noticed this.I am stilling trying to fix all these problems.
